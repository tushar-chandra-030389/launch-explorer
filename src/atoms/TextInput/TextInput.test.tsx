import { render, screen, fireEvent } from "@testing-library/react";
import TextInput from ".";

describe("TextInput", () => {
  it("should render the input field", () => {
    const { asFragment } = render(
      <TextInput label="Input label" name="testInput" />
    );
    const label = screen.getByLabelText("Input label");
    expect(label).not.toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should call the handler on input change", () => {
    const mockCallback = jest.fn();
    render(
      <TextInput label="Input label" name="testInput" onChange={mockCallback} />
    );
    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: "new input value" } });
    expect(mockCallback).toHaveBeenNthCalledWith(
      1,
      "testInput",
      "new input value"
    );
  });
});
