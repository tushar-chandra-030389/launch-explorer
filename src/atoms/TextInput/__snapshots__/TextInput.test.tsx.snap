// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render the input field 1`] = `
<DocumentFragment>
  <label
    class="block"
  >
    <span
      class="after:content-['*'] after:ml-0.5 after:text-red-500 block text-sm font-medium text-slate-50"
    >
      Input label
    </span>
    <input
      class="mt-1 px-3 py-2 bg-backgroundSecondary border shadow-sm border-slate-300 placeholder-slate-400 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1"
      placeholder=""
      type="text"
      value=""
    />
  </label>
</DocumentFragment>
`;
